pilot/pkg/serviceregistry/kube/conversion.go

```diff
@@ -113,6 +113,18 @@ func convertService(svc v1.Service, domainSuffix string) *model.Service {
 	}
 	sort.Strings(serviceaccounts)
 
+	var lbAddrs []string
+	if svc.Spec.Type == v1.ServiceTypeLoadBalancer && len(svc.Status.LoadBalancer.Ingress) > 0 {
+		lbAddrs = make([]string, len(svc.Status.LoadBalancer.Ingress))
+		for i, ingress := range svc.Status.LoadBalancer.Ingress {
+			if len(ingress.IP) > 0 {
+				lbAddrs[i] = ingress.IP
+			} else if len(ingress.Hostname) > 0 {
+				lbAddrs[i] = ingress.Hostname
```

Comment:
if you assign this mixed mode IP/hostname, it will break all EDS. An EDS cluster cannot have both ip and hostname

Reply:
fixed